### PR TITLE
feat: add OffsetDateXmlAdapter

### PR DESCRIPTION
**Pull request by tacascer**
This PR adds support for threeten-extra's [OffsetDate](https://www.threeten.org/threeten-extra/apidocs/org.threeten.extra/org/threeten/extra/OffsetDate.html).

Addresses #15

### CODE REVIEW

**Comment by raphw** [comment]
Note that the offset is optional. To support mixed representations, I suggested these: https://github.com/threeten-jaxb/threeten-jaxb/pull/22

Possibly you should add them, too?

**Comment by tacascer** [comment]
> Note that the offset is optional. To support mixed representations, I suggested these: #22

Seems sensible. I will switch to use the [ISO_DATE formatter](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/format/DateTimeFormatter.html?is-external=true#ISO_DATE) instead

File: threeten-jaxb-extra/src/test/java/io/github/threetenjaxb/extra/OffsetDateXmlAdapterTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    private static final Map<String, OffsetDate> STRING_OFFSET_DATE_MAP = new HashMap<>();
+
+    static {
+        STRING_OFFSET_DATE_MAP.put("2020-01-15+01:00", OffsetDate.of(2020, 1, 15, ZoneOffset.ofHours(1)));
+    }
+
+    OffsetDateXmlAdapterTest() {
+        super(new OffsetDateXmlAdapter(), STRING_OFFSET_DATE_MAP);
+    }
+}
```
Comment (haldokan) [inline threeten-jaxb-extra/src/test/java/io/github/threetenjaxb/extra/OffsetDateXmlAdapterTest.java:20]: eol

**Comment by tacascer** [comment]
Alright @raphw I've added the functionality for parsing dates without offsets

**Comment by tacascer** [comment]
Hey @mkroening I think it's ready for prime time.

Would you mind taking a look when you get the chance?

Thanks a lot!

**Review by mkroening — APPROVED** [review]
Thanks! :)

**Comment by mkroening** [comment]
I published this in [`v2.2.0`](https://github.com/threeten-jaxb/threeten-jaxb/releases/tag/v2.2.0). :)

**Pull request merged**